assets/src/blocks/Covers/useCovers.js

```diff
@@ -53,26 +54,38 @@ export const useCovers = ({ post_types, tags, cover_type, initialRowsLimit, post
     const path = addQueryArgs('planet4/v1/get-covers', args);
 
     try {
-      const loadedCovers = await apiFetch({ path });
+      const loadedCovers = await apiFetch( { path, signal: controller.signal } );
 
-      if (loadedCovers) {
+      if(loadedCovers) {
         setCovers(loadedCovers);
       }
-
-    } catch (e) {
-      console.log(e);
-      setError(e.message);
-    } finally {
-      setLoading(false);
+    } catch (err) {
+      setError(err.message);
     }
+
+    setLoading(false);
   };
 
   useEffect(() => {
     if (!noLoading) {
-      loadCovers();
+      setController(typeof AbortController === 'undefined' ? undefined : new AbortController());
     }
   }, [cover_type, post_types, tags, posts, layout]);
 
+  useEffect(() => {
+    if(controller) {
+      loadCovers();
+    }
+
+    return () => {
```

Comment:
This clean up function is invoked every time the component is being unmounted. You can take a look at the [React Documentation](https://reactjs.org/docs/hooks-effect.html#effects-with-cleanup) for more information.